test/instrumentation/main.go

```diff
@@ -173,3 +179,100 @@ func globalVariableDeclarations(tree *ast.File) map[string]ast.Expr {
 	}
 	return consts
 }
+
+func importedGlobalVariableDeclaration(localVariables map[string]ast.Expr, imports []*ast.ImportSpec) (map[string]ast.Expr, error) {
+	// to test the import will be rooted at GOPATH, so probably import will be something like /test/instrumentation/testpackage
+
+	// env gets
+	//GOMODCACHE := os.Getenv("GOMODCACHE")
+	GOROOT := os.Getenv("GOROOT")
+	GOOS := os.Getenv("GOOS")
+	KUBE_ROOT := os.Getenv("KUBE_ROOT")
+	KUBE_URL_ROOT := "k8s.io/kubernetes"
+	KUBE_ROOT_INTERNAL := strings.Replace(KUBE_ROOT, KUBE_URL_ROOT, "", 1) // k8s/k8s refs need this stripped
+	
+	for _, im := range imports {
+		// get imported label
+		importAlias := "unknown"
+		if im.Name == nil {
+			pathSegments := strings.Split(im.Path.Value, "/")
+			importAlias = strings.Trim(pathSegments[len(pathSegments)-1], "\"")
+		} else {
+			importAlias = im.Name.String()
+		}
+
+		// parse directory path
+		pathPrefix := "unknown"
+		if strings.Contains(im.Path.Value, KUBE_URL_ROOT) {
+			// search k/k local checkout
+			pathPrefix = KUBE_ROOT_INTERNAL
+		} else if strings.Contains(im.Path.Value, "k8s.io/") {
+			// search k/k/staging local checkout
+			pathPrefix = KUBE_ROOT + string(os.PathSeparator) + "staging" + string(os.PathSeparator) + "src"  //KUBE_ROOT + string(os.PathSeparator) + "vendor" //
+		} else if strings.Contains(im.Path.Value, ".") {
+			// not stdlib -> prefix with GOMODCACHE
+			// pathPrefix = KUBE_ROOT + string(os.PathSeparator) + "vendor"
+
+			//  this requires implementing SIV, skip for now
+			continue
+		} else {
+			// stdlib -> prefix with GOROOT
+			pathPrefix = GOROOT + string(os.PathSeparator) + "src"
```

Comment:
why not path.join?

Reply:
will do.

Reply:
all string concats have been changed to strings.Join